### PR TITLE
Metrics are obsolete

### LINKED ISSUE #1370: Mark metrics related API as "experimental"
The current Metric API/SDK implementation is a prototype based on initial draft version of the Otel Metrics spec. This will be completely rewritten to match the new spec. This rewrite is expected to happen [after](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1501) the 1.0.0 release of Trace/Logs components. Since Metrics is also shipped as part of the same package as Trace/Logs, we need to let users know that Metrics is a preview version only.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1370.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/CHANGELOG.md
Comment (alanwest) [inline src/OpenTelemetry.Api/CHANGELOG.md:11]: I like the description you provide with the `Obsolete` attribute. I think providing the same info in the changelog could be useful. Something like:

> Metrics API/SDK support is in an experimental state and is not recommended for production use. All metric APIs have been marked with the `Obsolete` attribute. See https://github.com/open-telemetry/opentelemetry-dotnet/issues/1501 for more information.

Comment (eddynaka) [reply src/OpenTelemetry.Api/CHANGELOG.md]: thank you for reviewing. just updated!

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (cijothomas) [inline src/OpenTelemetry/OpenTelemetry.csproj:8]: Will open a new issue to move CS0618 nowarn to individual metric related files, not for whole project.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
Okay to merge once we get 2 approvals.

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (cijothomas) [reply src/OpenTelemetry/OpenTelemetry.csproj:8]: @Austin-Tan Can you work on this? (a separate PR)

**Review by alanwest — APPROVED** [review]
LGTM

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (Austin-Tan) [reply src/OpenTelemetry/OpenTelemetry.csproj:8]: Sure I'll get started right now. Would you recommend just Ctrl+F'ing the project for references to components in src/OpenTelemetry/Metrics, and appending `#pragma warning disable cs0618` to the top?
And double checking my understanding - why don't we want compiler warnings? Aren't we intending to make it clear that Metrics are experimental?

Comment (cijothomas) [reply src/OpenTelemetry/OpenTelemetry.csproj:8]: Metrics are marked Obsolete so customers should be alerted. But there are pieces in this repo itself which depends on the metric, so they should be excluded from this warning.

(current its a project level warning exclusion, which will prevent us from catching any real issues, outside of metrics).

**Pull request merged**
